benchmarks/nemo/nemolite2d/manual_versions/psykal_opencl/allkernels_tasks.cl

```diff
@@ -87,8 +163,12 @@ __kernel void bc_solid_u_code(
   const __global int * restrict tmask
   ){
   for (int jj = ystart; jj <= ystop; jj++){
+      double this_tmask;
+      double next_tmask = tmask[jj * LEN + xstart];
       for (int ji = xstart; ji <= xstop; ji++){
-          if (((tmask[jj * LEN + ji] * tmask[jj * LEN + (ji + 1)]) == 0)) {
+          this_tmask = next_tmask;
+          next_tmask = tmask[jj * LEN + (ji + 1)];
+          if (((this_tmask * next_tmask) == 0)) {
```

Comment:
Ah you are right, these are integer values. I will leave the (now integer) multiplication instead of the 2 comparisons for now as its the same as the original kernel.